spyder_kernels/customize/spydercustomize.py

```diff
@@ -740,6 +740,8 @@ def runcell(cellname, filename):
         pass
     run_umr()
     ipython_shell = get_ipython()
+    namespace = _get_globals()
```

Comment:
@ccordoba12 Should I just use `global __file__` here instead? If that works. I haven't tested it.

Reply:
No. I think the way you implemented this is the right way to do it.